src/components/VariablesTable.vue

```diff
@@ -3,24 +3,24 @@
     <p-layout-table sticky>
       <template #header-start>
         <div class="variables-table__header-start">
+          <VariablesDeleteButton v-if="can.delete.variable" :variable-ids="selectedVariables" @delete="deleteVariables" />
           <ResultsCount v-if="selectedVariables.length == 0" :label="localization.info.variable" :count="variablesCount" />
           <SelectedCount v-else :count="selectedVariables.length" />
-
-          <FlowsDeleteButton v-if="can.delete.variable" :selected="selectedVariables" @delete="deleteVariables" />
         </div>
       </template>
 
       <template #header-end>
         <div class="variables-table__header-end">
           <SearchInput v-model="variableLike" :placeholder="localization.info.variablesSearch" :label="localization.info.variablesSearch" />
           <p-select v-model="filter.sort" :options="variableSortOptions" />
-          <p-tags-input v-model="filter.variables.tags.name" :empty-message="localization.info.tags" class="variables-table__tags" />
+          <p-tags-input v-model="filter.variables.tags.name" :empty-message="localization.info.filterByTags" class="variables-table__tags-input" />
         </div>
       </template>
 
       <p-table :data="variables" :columns="columns">
         <template #selection-heading>
-          <p-checkbox v-model="model" @update:model-value="selectAllVariables" />
+          <p-checkbox v-if="variables.length" v-model="model" @update:model-value="selectAllVariables" />
+          <div v-else />
```

Comment:
This new check on `variables.length` and the empty div ensure the checkbox isn't shown on an empty table and that the `selection` heading isn't shown when the checkbox is hidden (an empty slot will render the column label by default)